qiskit_nature/properties/second_quantization/electronic/integrals/electronic_integrals_utils/_electronic_integrals_utils.py

```diff
@@ -0,0 +1,118 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utility functions to detect and transform the index-ordering convention of two-body integrals"""
+
+import numpy
+
+## These are the permutation symmetries satisfied by
+## a rank-4 tensor of real two-body integrals in chemists'
+## index order.
+## HJO is Molecular Electronic Structure Theory by Helgaker, Jørgensen, Olsen
+CHEM_INDEX_PERMUTATIONS = [
+    "pqrs->qprs",  # (1.4.38) HJO (1)
+    "pqrs->pqsr",  # (1.4.38) HJO (2)
+    "pqrs->qpsr",  # (1.4.38) HJO (3)
+    "pqrs->rspq",  # (1.4.17) HJO (4)
+    "pqrs->rsqp",  # 1 and 4
+    "pqrs->srpq",  # 2 and 4
+    "pqrs->srqp",  # 3 and 4
+]
+
+
+def phys_to_chem(two_body_tensor):
+    """
+    Convert the rank-four tensor `two_body_tensor` representing two-body integrals from physicists'
+    index order to chemists' index order: i,j,k,l -> i,l,j,k
+
+    See `chem_to_phys`, `check_two_body_symmetries`.
+    """
+    permuted_tensor = numpy.einsum("ijkl->iljk", two_body_tensor)
+    return permuted_tensor
+
+
+def chem_to_phys(two_body_tensor):
+    """
+    Convert the rank-four tensor `two_body_tensor` representing two-body integrals from chemists'
+    index order to physicists' index order: i,j,k,l -> i,k,l,j
+
+    See `phys_to_chem`, `check_two_body_symmetries`.
+
+    Note:
+    Denote `chem_to_phys` by `g` and `phys_to_chem` by `h`. The elements `g`, `h`, `I` form
+    a group with `gh = hg = I`, `g^2=h`, and `h^2=g`.
+    """
+    permuted_tensor = numpy.einsum("ijkl->iklj", two_body_tensor)
```

Comment:
This numpy einsum exists as `TwoBodyElectronicIntegrals.EINSUM_CHEM_TO_PHYS`. I think it would make sense to refactor the `TwoBodyElectronicIntegrals` to purely make use of the utilities here instead.